src/drive/web/modules/actionmenu/MenuItem.jsx

```diff
@@ -1,30 +1,22 @@
-import React from 'react'
-import { connect } from 'react-redux'
-import Toggle from 'cozy-ui/react/Toggle'
+import React, { Component } from 'react'
 
-import { isAvailableOffline } from 'drive/web/modules/drive/availableOffline'
-
-const MenuItem = ({ className, children, checkbox, onClick }) => (
-  <div>
-    <a className={className} onClick={onClick}>
-      {children}
-      {checkbox && (
-        <Toggle
-          id={children}
-          checked={checkbox.value}
-          onToggle={checkbox.onChange}
-        />
-      )}
-    </a>
-  </div>
-)
-export default MenuItem
-
-const mapStateToProps = (state, ownProps) => ({
-  checkbox: {
-    value: isAvailableOffline(state)(ownProps.files[0].id),
-    onChange: () => {}
+export default class MenuItem extends Component {
+  onClick = e => {
+    // we need to stop propagation here so that the menu doesn't close itself
+    e.stopPropagation()
+    if (this.props.onClick) {
+      this.props.onClick()
+    }
```

Comment:
Linking this for reference, a bit late : https://github.com/cozy/cozy-ui/pull/571

Rather than calling `e.stopPropagation()` all the time and `onClose` most of the time, I think it'd be better to let the menu close itself by default all the time, and when we don't want to close the menu — in our case, only when using the offline toggle — we can call `e.preventDefault()`, and cozy-ui can take care of stopping the propagation.

Not blocking because it's ok if we do it later, but it'd be a nice chance :)

Reply:
If you think your PR can be merged soon, why not?

Reply:
Because it doesn't actually do what I describe in the last message yet :D

Reply:
Oh I see... Too bad then :smile: